src/v2/wallet.ts

```diff
@@ -1091,11 +1097,27 @@ Wallet.prototype.prebuildTransaction = function(params, callback) {
  * @return {*}
  */
 Wallet.prototype.signTransaction = function(params, callback) {
-  const userKeychain = params.keychain || params.key;
   const txPrebuild = params.txPrebuild;
   if (!txPrebuild || typeof txPrebuild !== 'object') {
     throw new Error('txPrebuild must be an object');
   }
+  const userPrv = this.getUserPrv(params);
```

Comment:
I'd rather decrypt the user prv only after presignTransaction has been called (eg. on line 1109), so we already know the tx is good to sign. We should keep the userPrv around in the clear for only as long as it is absolutely required.

Reply:
Makes sense, done